fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java

```diff
@@ -98,15 +99,16 @@ public static void checkIsLakeTable(String dbName, String tableName) throws Anal
     }
 
     public static void checkPartitionValuesExistForReplaceListPartition(ListPartitionInfo listPartitionInfo,
-                                                                        Partition partition)
+                                                                        Partition tempPartition)
             throws DdlException {
         try {
             List<Long> partitionIds = listPartitionInfo.getPartitionIds(false);
```

Comment:
should only check items in `partitionNames` that will be replaced.

and also consider `strictRange` like  range partition replace?